compiler/src/iree/compiler/Pipelines/Pipelines.cpp

```diff
@@ -41,6 +41,7 @@ void buildIREEVMTransformPassPipeline(
       buildTOSAInputConversionPassPipeline(passManager);
       break;
     case InputDialectOptions::Type::mhlo:
+      MHLO::buildMHLOCleanupPassPipeline(passManager);
```

Comment:
do we need a separate pipeline? Can we just add it to the existing pipeline. That way it will work with the `iree-opt <input-conversion-pipeline-flag> <convertion-to-flow-pipeline-flag>... ` that is also useful (and used by few folks on the team)

Reply:
`iree-opt --iree-transformation-pipeline --iree-input-type=mhlo ...` would still work I believe (but didn't test). I considered just adding it to `buildMHLOInputConversionPassPipeline` but that would just do redundant work for XLA (which I'm not too worried about as that path would be used less and less). I can register one additional to XLA cleanup pipeline registration or add it to the other, WDYT?

Reply:
Ah I see its mirroring what is happening with `xla`. Do you mind trying what would happen if you moved the passes in `buildXLACleanupPassPipeline` into `buildMHLOInputConversionPassPipeline`. Then the MHLO and XLA path would be the same (dont know why they are different actually, maybe legacy).
In any case, this is fine for me. Stamping this. Fine with me if the merging of XLA/MHLO pipelines is done after submit of this PR.